atomic_reactor/plugins/pre_bump_release.py

```diff
@@ -0,0 +1,263 @@
+    from shlex import quote
+except ImportError:
+    from pipes import quote
+
+import os
+from pygit2 import init_repository, Signature
+import subprocess
+
+from atomic_reactor.plugin import PreBuildPlugin
+from atomic_reactor.source import GitSource
+from atomic_reactor.plugins.pre_check_and_set_rebuild import is_rebuild
+from atomic_reactor.constants import PY2
+from dockerfile_parse import DockerfileParser
+
+
+def shlex_splits(value):
+    if PY2 and isinstance(value, unicode):
+        value = value.encode('utf-8')
+        splits = shlex.split(value)
+        return [split.decode('utf-8') for split in splits]
+    else:
+        return shlex.split(value)
+
+
+class BumpReleasePlugin(PreBuildPlugin):
+    """
+    Git branch management plugin
+
+    For rebuilds, push a new commit incrementing the Release label.
+
+    For initial builds, verify the branch is at the specified commit
+    hash.
+
+    Example configuration:
+
+    {
+      "name": "bump_release",
+      "args": {
+        "git_ref": "12345678....",
+        "author_name": "OSBS Build System",
+        "author_email": "root@example.com"
+      }
+    }
+
+    Additional optional arguments:
+    - committer_name
+    - committer_email
+    - commit_message
+    - push_url
+    """
+
+    key = "bump_release"
+    can_fail = False  # We really want to stop the process
+
+    commit_message = "Bumped release for automated rebuild"
+
+    def __init__(self, tasker, workflow,
+                 git_ref,
+                 author_name, author_email,
+                 committer_name=None, committer_email=None,
+                 commit_message=None,
+                 push_url=None):
+        """
+        constructor
+
+        :param tasker: DockerTasker instance
+        :param workflow: DockerBuildWorkflow instance
+        :param git_ref: str, commit hash expected on first build
+        :param author_name: str, name to use for git commits
+        :param author_email: str, email address for git commits
+        :param committer_name: str, name to use for git commits (else author's)
+        :param committer_email: str, email address for git commits (else
+                                     author's)
+        :param commit_message: str, git commit message
+        :param push_url: str, URL for push
+        """
+        # call parent constructor
+        super(BumpReleasePlugin, self).__init__(tasker, workflow)
+        self.git_ref = git_ref
+        self.author_name = author_name
+        self.author_email = author_email
+        self.committer_name = committer_name
+        self.commiter_email = committer_email
+        if committer_name is None:
+            self.committer_name = author_name
+        if committer_email is None:
+            self.committer_email = author_email
+
+        self.push_url = push_url
+        if commit_message is not None:
+            self.commit_message = commit_message
+
+    def bump_local_file(self, parser, label_key, next_release):
```

Comment:
You missed it because the code is much larger than it looks like it ought to be when it's getting so much help. :-)

Perhaps some of this could be moved to `dockerfile_parse`.